### PR TITLE
Support mouse buttons/wheel in macros

### PR DESCRIPTION
**Pull request by roxya**
- Wheel/middle click
- Wheel Up/Down
- X1/X2

### CODE REVIEW

File: src/Game/Scenes/GameSceneInputHandler.cs [hunk 1]
```diff
@@ -921,6 +945,21 @@ internal override bool OnMouseWheel(bool up)
                 }
             }
 
+            if (CanExecuteMacro())
+            {
+                Macro macro = Macros.FindMacro(up, Keyboard.Alt, Keyboard.Ctrl, Keyboard.Shift);
+
+                if (macro != null)
+                {
+                    if (macro.Items != null && macro.Items is MacroObject mac)
```
Comment (andreakarasho) [inline src/Game/Scenes/GameSceneInputHandler.cs]: `if (macro.Items is MacroObject mac)`
is enough, implicitly check for nullptr

File: src/Game/Scenes/GameSceneInputHandler.cs [hunk 2]
```diff
@@ -902,6 +906,26 @@ private bool OnRightMouseDoubleClick()
...
 
+        private bool OnExtraMouseDown(MouseButtonType button)
+        {
+            if (CanExecuteMacro())
+            {
+                Macro macro = Macros.FindMacro(button, Keyboard.Alt, Keyboard.Ctrl, Keyboard.Shift);
+
+                if (macro != null && button != MouseButtonType.None)
+                {
+                    if (macro.Items != null && macro.Items is MacroObject mac)
```
Comment (andreakarasho) [inline src/Game/Scenes/GameSceneInputHandler.cs]: `if (macro.Items is MacroObject mac)`
is enough, implicitly check for nullptr

File: src/Game/Scenes/GameSceneInputHandler.cs (hunk 1 above)
Comment (roxya) [reply src/Game/Scenes/GameSceneInputHandler.cs]: that's what I get for copying the existing code! fixed, thanks

File: src/Game/Scenes/GameSceneInputHandler.cs (hunk 2 above)
Comment (roxya) [reply src/Game/Scenes/GameSceneInputHandler.cs]: fixed, thanks

**Pull request merged**